### PR TITLE
feat: [FileCollection] add function to retain multiple patterns

### PR DESCRIPTION
**Pull request by christianberkman**
**Description**
Supersedes #8940

Add a method to ``FileCollection`` that accepts an array of strings to retain multiple patterns in the file collection.

**Checklist:**
- [X] Securely signed commits
- [X] Component(s) with PHPDoc blocks, only if necessary or adds value
- [X] Unit testing, with >80% coverage
- [X] User guide updated
- [X] Conforms to style guide

### CODE REVIEW

File: user_guide_src/source/changelogs/v4.6.0.rst [hunk 1]
```diff
@@ -66,6 +66,9 @@ Removed Deprecated Items
 Enhancements
 ************
 
+- Added `retainMultiplePatterns()` to `FileCollection` class.
```
Comment (kenjis) [inline user_guide_src/source/changelogs/v4.6.0.rst:69]: You are adding it in the wrong place.
`FileCollection` is a library, so please add in the [`Libraries` section](https://github.com/codeigniter4/CodeIgniter4/blob/61dc880edc560ead12a28a3713160043930f2747/user_guide_src/source/changelogs/v4.6.0.rst#libraries}

**Comment by kenjis** [comment]
PHPStan and Rector checks in GitHub Actions failed.
Please fix them.

File: system/Files/FileCollection.php [hunk 2]
```diff
@@ -340,6 +340,44 @@ public function retainPattern(string $pattern, ?string $scope = null)
         return $this->removeFiles(array_diff($files, self::matchFiles($files, $pattern)));
     }
 
+    /**
+     * Keeps only the files from the list that match multiple patterns
+     * (within the optional scope).
+     *
+     * @param array<string>       $patterns Array of regex or pseudo-regex strings
```
Comment (kenjis) [inline system/Files/FileCollection.php:347]:
```suggestion
     * @param list<string>       $patterns Array of regex or pseudo-regex strings
```
and run `composer cs-fix` to fix coding style.

**Comment by kenjis** [comment]
@christianberkman You are using the PR branch name `4.6`.
It is a bad practice. I cannot checkout your PR branch with `gh pr checkout 8953` command, because `4.6` is the branch name already used in this main repository.
Please name another branch name like `feat-something` in the future.

**Comment by christianberkman** [comment]
My apologies and thank you for the clarification. Is there any use in renaming the branch now or do I start another PR with the renamed branch? Three times is the charm...

**Comment by kenjis** [comment]
There is no way to change the PR branch.
Feel free to send another PR.

**Pull request closed without merging**